### PR TITLE
Corrections redux

### PR DESCRIPTION
**Pull request by roberbnd**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (Spawbreaker) [inline .vscode/launch.json]: Is this file necessary? I think it's VS auto generated, so you should not include it.

File: src/api/authenticationApi.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      try {
+        await fetchToken(data, admin).then(
+          response => {
+            if(response.status === 200) {
+              dispatch(authenticatedAction(response))
+            } else {
+              dispatch(authentication_error(response))
+            }
+          },
+          error => console.log('An error accurred', error))
```
Comment (Spawbreaker) [inline src/api/authenticationApi.js]: Ocurred*

File: src/components/login/login.jsx [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
+  const dispach = useDispatch();
   const [loginState, setLoginState] = useState({ email: "", password: "" });
 
   const Send = event => {
     event.preventDefault();
     let formData = new FormData()
     formData.set('email', loginState.email)
     formData.set('password', loginState.password)
-    props.login(formData)
+    dispach(login(formData, props.admin))
```
Comment (Spawbreaker) [inline src/components/login/login.jsx]: Why are you sending props.admin ? What does props.admin stand for?

File: src/containers/layout.js [hunk 4]
```diff
@@ -1,63 +1,70 @@
...
+   if(accessLevelState.role === "admin"){
+    layout = (
+      <LayoutAdmin accessLevelState = {accessLevelState}>
+        <Switch> 
+          <Route path = "/admin" exact component={WorkInProgress} /> 
+          <Route path = "/admin/products" exact component={WorkInProgress} />
+          <Route path = "/admin/orders" exact component={WorkInProgress} />
+          <Route path = "/admin/staff" exact component={WorkInProgress} />
+          <Route path = "/admin/banners" exact component={WorkInProgress} />
+          <Route path = "/*" exact component={WorkInProgress} />
```
Comment (Spawbreaker) [inline src/containers/layout.js]: Shouldn't this one be ` exact component={NotFound}` ?

File: src/reducers/authenticationReducer.js [hunk 5]
```diff
@@ -5,25 +5,37 @@ import {
...
+  refresh_token: null 
 };
 
+const autenticationTypes = {
+  [AUTHENTICATED]: (newState, data) => {
+    newState.name = (
+      data.customer_name ? data.customer_name : 
+      data.admin_name ? data.admin_name : newState.name
+    )
+    newState.role = ( data.message === "Welcome admin" ? "admin" :  "registeredUser" )
```
Comment (Spawbreaker) [inline src/reducers/authenticationReducer.js]: Shouldn't we also greet the admin with their name? Maybe this is not the right way to ask for this. It will break the authentication if we change the greeting message to anything else, like, "Greetings admin", "Good morning admin" or anything.

File: src/store/index.js [hunk 6]
```diff
@@ -3,5 +3,7 @@ import rootReducer from '../reducers';
 import thunk from 'redux-thunk';
 
 const store = createStore(rootReducer, applyMiddleware(thunk))
+//const store = createStore(rootReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())
+//window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
```
Comment (Spawbreaker) [inline src/store/index.js]: Unecessary

File: src/containers/layout.js (hunk 4 above)
Comment (pablorich) [reply src/containers/layout.js]: also, the incongruity on a /* with the exact keyword

File: src/store/index.js (hunk 6 above)
Comment (christianahvilla) [reply src/store/index.js]: Is necessary to debug redux, only for development

File: src/components/layout/navigation-items/navigation-items.js [hunk 7]
```diff
@@ -17,7 +17,7 @@ const navigationItems = (props) => {
                 <Item link="/admin/orders">Orders</Item>
                 <Item link="/admin/staff">Staff</Item>
                 <Item link="/admin/banners">Banners</Item>
-                <Item>Log out</Item>
+                <Item link="/admin/logout">Log out</Item>
```
Comment (pablorich) [inline src/components/layout/navigation-items/navigation-items.js]: Logout isn't a view, so it doesn't use a link.
Instead it calls a function to terminate the session with the API and then redirect to homepage or login, idk.
```
const createOrder = () =>{
        setIsLoading({
            loading: true
        });

        axios.post(createOrderUrl, {}, defaultHeaders)
        .then(response => {
            setIsLoading({
                loading: false
            });
            props.history.push('/checkout')
        })
        .catch(error=>{
            setIsLoading({
                loading: false
            })
        });
    }

```
This is code I used in cart-view to make an API call and then redirect

Comment (AnaSandoval-Intersys) [reply src/components/layout/navigation-items/navigation-items.js]: This one's on me. If we didn't include it as a link it threw a warning (because item is expecting a link) that made travis reject it.
Since we reaaaally need this PR to start working with sending requests with the token, leaving it as a view is a quick fix until the behavior of the log out is implemented, which will be in a different pull request and hence it will take more time. I'd suggest to leave it like this now so we can start working, and someone can separately work in the logout.

File: src/components/login/login.jsx (hunk 3 above)
Comment (roberbnd) [reply src/components/login/login.jsx]: it is the way to know if the login component is in the view /login-admin or /login then we have
<Login admin /> or just <Login />

<Login admin /> with this props we know that axios needs to send  request in this line "request().post('admin/session/', data)"

File: src/reducers/authenticationReducer.js (hunk 5 above)
Comment (roberbnd) [reply src/reducers/authenticationReducer.js]: image

it has admin's greetings

File: src/containers/layout.js (hunk 4 above)
Comment (roberbnd) [reply src/containers/layout.js]: Done
